src/screencast/pipewire_screencast.c

```diff
@@ -166,27 +166,32 @@ void xdpw_pwr_stream_init(struct xdpw_screencast_instance *cast) {
 	enum spa_video_format format = xdpw_format_pw_from_wl_shm(cast);
 	enum spa_video_format format_without_alpha =
 		xdpw_format_pw_strip_alpha(format);
-	uint32_t n_formats = 1;
+
+	struct spa_pod_frame f;
+	spa_pod_builder_push_object(&b, &f, SPA_TYPE_OBJECT_Format, SPA_PARAM_EnumFormat);
+	spa_pod_builder_add(&b, SPA_FORMAT_mediaType, SPA_POD_Id(SPA_MEDIA_TYPE_video), 0);
```

Comment:
Would it make more sense to replace these trailing `0` with `NULL`?

Reply:
I've put `0` because that's what it is inside the macro that I expanded, but sure I can change it

Reply:
in the pipewire [docs](https://gitlab.freedesktop.org/pipewire/pipewire/-/blob/master/doc/spa/pod.md) they use `0` though

Reply:
Ah, actually you were correct, 0 is better. `spa_pod_builder_add` accepts an arbitrary number of `SPA_FORMAT_*` enum values, and zero ends the list. Since enum values are integers, 0 makes more sense.

Reply:
no problem